BBC-archive/psammead
# New inline link styling

Resolves #397
Resolves #716

**Overall change:** Adds the new UX inline link styling which is a background colour on hover. Also maintains the border changing from 1px to 2px on hover for users in high constrast modes.

**Code changes:**

- On hover/focus the background becomes red and the text becomes white
- On hover/focus adds `pre-wrap` so that for links that wrap to new lines the letter on the end of the line has a slight background buffer
- Maintains a 2px border on hover/focus so there is a visual state change on high contrast modes
- On hover/focus adds minus margin and padding to increase the background for buffer between the end of the background colour and the actual letter (see screenshots with and without in comments)

Testing notes:

This is most easily tested by following the instructions on https://github.com/bbc/psammead/pull/719 and then testing as described in the notes below. Happy to pair with test on this work. 

- Open storybook and regression test the new inline link styling using the caption and paragraph stories that contain inline links, make sure to check different world services with varying scripts and line-heights. NB: The reason we should not use the inline-link story for this regression is that it does not set it's own typography styling as it's expected to be used as a child of another component
- Check firefox high contrast mode (Go to FireFox -> Preference -> Language & Appearance -> Colours -> 'Override the colours ...' set as Always)
- Check Windows high contrast mode
- Check with page zoomed to high percentages (300%+)
- Check across different breakpoints
- Check across different browsers
- Check #397 is fixed as part of this
---
